From 001dd9b829eb8e8b4465d87c6c70ccf4ee2e6b6a Mon Sep 17 00:00:00 2001
From: Silvio Traversaro <silvio@traversaro.it>
Date: Wed, 14 Jun 2023 07:20:11 +0200
Subject: [PATCH] Protobuf: Do not require version 3 do support Protobuf 4.23.2
 (23.2) (#2006)

Signed-off-by: Silvio Traversaro <silvio@traversaro.it>

* TriggeredPublisher: don't catch FatalException
  It has been removed from recent versions of protobuf.

Signed-off-by: Steve Peters <scpeters@openrobotics.org>
Co-authored-by: Steve Peters <scpeters@openrobotics.org>
---
 CMakeLists.txt                                        | 8 +++-----
 src/systems/triggered_publisher/TriggeredPublisher.cc | 9 +--------
 2 files changed, 4 insertions(+), 13 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 77208a11eb..5644db73f7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -186,12 +186,10 @@ set(GZ_UTILS_VER ${gz-utils2_VERSION_MAJOR})
 
 #--------------------------------------
 # Find protobuf
-set(REQ_PROTOBUF_VER 3)
 gz_find_package(GzProtobuf
-                 VERSION ${REQ_PROTOBUF_VER}
-                 REQUIRED
-                 COMPONENTS all
-                 PRETTY Protobuf)
+                REQUIRED
+                COMPONENTS all
+                PRETTY Protobuf)
 set(Protobuf_IMPORT_DIRS ${gz-msgs9_INCLUDE_DIRS})
 
 #--------------------------------------
diff --git a/src/systems/triggered_publisher/TriggeredPublisher.cc b/src/systems/triggered_publisher/TriggeredPublisher.cc
index 54b3756d4b..87d0add654 100644
--- a/src/systems/triggered_publisher/TriggeredPublisher.cc
+++ b/src/systems/triggered_publisher/TriggeredPublisher.cc
@@ -832,14 +832,7 @@ bool TriggeredPublisher::MatchInput(const transport::ProtoMsg &_inputMsg)
   return std::all_of(this->matchers.begin(), this->matchers.end(),
                      [&](const auto &_matcher)
                      {
-                       try
-                       {
-                         return _matcher->Match(_inputMsg);
-                       } catch (const google::protobuf::FatalException &err)
-                       {
-                          gzerr << err.what() << std::endl;
-                          return false;
-                       }
+                       return _matcher->Match(_inputMsg);
                      });
 }

